src/LifeCycle.js

```diff
@@ -0,0 +1,62 @@
+export default function doLifeCycle(main, nextValue, opts = {}) {
+  let transform;
+  let beforeUpdate;
+  let afterUpdate;
+
+  const defaults = {
+    transform: value => value,
+    beforeUpdate: [() => true],
+    afterUpdate: [],
+  };
+
+  if (typeof opts === 'function') {
+    transform = opts;
+    ({ beforeUpdate, afterUpdate } = defaults);
+  } else if (typeof opts === 'object') {
+    if (opts.beforeUpdate || opts.afterUpdate) {
+      const options = [...defaults, ...opts];
+
+      transform = options.transform ? options.transform : value => value;
+      beforeUpdate = [() => true].concat(options.beforeUpdate);
```

Comment:
not sure how I feel about always having a beforeUpdate. 

imo, we should just not loop/call beforeUpdates if `beforeUpdate` is empty.